Gems/GradientSignal/Code/Tests/GradientSignalTest.cpp

```diff
@@ -156,10 +154,16 @@ namespace UnitTest
         config.m_amplitude = 3.0f;
         config.m_frequency = 1.13f;
 
-        AZStd::vector<float> expectedOutput = { AZ_TRAIT_UNIT_TEST_PERLINE_GRADIANT_GOLDEN_VALUES_7878 };
+        AZStd::vector<float> expectedOutput = 
```

Comment:
It looks like the test has become "verify that the mock contains mock values" as well as incidentally verifying that a gradient component can be constructed, configured, and applied to another component. It's unclear to me what is entailed in a `PerlinGradientComponent_GoldenTest` based on its name, but I assume it should be expecting something specific about the functionality of the component. Adjacent tests appear to be directly verifying the math, so mocking out the calculation appears to defeat the intent of the test.

Reply:
Essentially my feedback is "this doesn't fix the underlying problem, it is a workaround which is not obviously labeled as a workaround"

Reply:
Well the underlying problem for the test is that it was relying on deterministic values based on random seeding and it’s outcome, but it was test vectors were based on a single platform. The problem is fixed by eliminating the random shuffling which was different per platform through the different levels of mocking. 
If there was a way to mock the std random function in prepare table in C++ then that would be the ideal here. 

The other option is to refactor that class to make it mock able but I am avoiding making changes to that production code unless the original author is okay with it